packages/pipeline/src/Pipeline.tsx

```diff
@@ -0,0 +1,275 @@
+import React, {
+  forwardRef,
+  createRef,
+  useState,
+  Ref,
+  ReactElement,
+  ReactNode,
+} from 'react';
+
+import PropTypes from 'prop-types';
+import { findLast } from 'lodash';
+
+import { uiColors } from '@leafygreen-ui/palette';
+import Tooltip from '@leafygreen-ui/tooltip';
+import { css, cx } from '@leafygreen-ui/emotion';
+import { useElementNode, useMutationObserver } from '@leafygreen-ui/hooks';
+
+import Stage from './Stage';
+import Counter from './Counter';
+
+import {
+  getPipelineCounterTooltip,
+  isStageElement,
+  isElementOverflowed,
+} from './utils';
+
+import { getRootStyle, Size, Variant, layout, colors } from './styles';
+
+interface StateForStyles {
+  hasHiddenStages: boolean;
+  size: Size;
+  variant: Variant;
+}
+
+interface PipelineProps {
+  /**
+   * Content that will appear inside of the Pipeline component.
+   */
+  children: ReactNode;
+
+  /**
+   * Optional className prop to apply to Pipeline content container.
+   */
+  className?: string;
+
+  /**
+   * Alter the rendered size of the component
+   */
+  size: Size;
+
+  /**
+   * Alter the visual apperance of the component
+   */
+  variant: Variant;
```

Comment:
It looks like we don't actually support multiple variants at the moment. If that's the case as it appears it is, it would significantly clean up the component to not support the prop at all.

Reply:
I would prefer the style implementation as is, as this defines the foundations for adding further variants should we need to, and helps keep the visual colours organised.

What I can do is remove the prop definition from the `propTypes` so that this cannot be changed at this point if that is more desirable?

Reply:
> In this case we have quite a lot of code that could be eliminated without the prop, and AFAIK we don't have any other components that has a variant prop when there's only a single variant.
> 
> While I know this makes adding variants easier in the future, is there any variant that's currently planned to be added right now? If not, I feel that this creates unnecessary overhead when reading the code, and should be removed.

This is starting to feel like bikeshedding. Resolved by b6c1bc0.